sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/RowEncoder.scala

```diff
@@ -137,6 +137,22 @@ object RowEncoder {
         CreateStruct(convertedFields))
   }
 
+  private def externalDataTypeForInput(dt: DataType): DataType = dt match {
```

Comment:
Now that there are several types of "get `DataType` methods` it would be good to make sure they are documents (specifically when should you use one over the other).